src/formatting_stack/linters/line_length.clj

```diff
@@ -2,29 +2,43 @@
   (:require
    [clojure.string :as string]
    [formatting-stack.protocols.linter :as linter]
-   [formatting-stack.util :refer [ensure-sequential process-in-parallel!]]
-   [nedap.utils.modular.api :refer [implement]]))
+   [formatting-stack.util :refer [ensure-sequential partition-between process-in-parallel!]]
+   [nedap.utils.modular.api :refer [implement]]
+   [medley.core :refer [assoc-some]]))
 
-(defn exceeding-lines [threshold filename]
-  (->> (-> filename slurp (string/split #"\n"))
-       (map-indexed (fn [i row]
-                      (let [column-count (count row)]
-                        (when (< threshold column-count)
-                          {:filename filename
-                           :source   :formatting-stack/line-length
-                           :level    :warning
-                           :column   column-count
-                           :line     (inc i)
-                           :msg      (str "Line exceeding " threshold " columns.")}))))
-       (filterv some?)))
+(defn exceeding-lines [{:keys [max-line-length merge-threshold]} filename]
+  (->> (-> filename slurp string/split-lines)
+       (map count)
+       (keep-indexed (fn [i column]
+                       (when (< max-line-length column)
+                         {:column column
+                          :line   (inc i)})))
```

Comment:
Sneaky. But why do we need to increment line by one :').

Reply:
Ah, okay, we are creating like a report here by looping over the lines, and zero indexed ah yeah.

Reply:
Yeah the collection is zero-indexed; most (all?) editors are one-indexed. so compensating for that here.